py3status/modules/volume_status.py

```diff
@@ -286,17 +293,18 @@ def deprecate_function(config):
         }
 
     def post_config_hook(self):
-        # Guess command if not set
+        # get a command
         if self.command is None:
             self.command = self.py3.check_commands(
-                ['amixer', 'pamixer', 'pactl']
-            )
+                ['amixer', 'pamixer', 'pactl'])
 
-        # device sometimes is an integer but should be passed to commands as a
-        # str.  So fix it here.
+        # turn integers to strings
+        if self.card is not None:
+            self.card = str(self.card)
         if self.device is not None:
-            self.device = '%s' % self.device
+            self.device = str(self.device)
```

Comment:
@lasers please be very careful changing things just because you can.  Although these may seem to do the same they are different.  `'%s' % self.device` is python2 friendly where as `str(self.device)` is not 

```
$ python
Python 2.7.13 (default, Sep  5 2017, 08:53:59) 
[GCC 7.1.1 20170622 (Red Hat 7.1.1-3)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> x = u'²'
>>> x
u'\xb2'
>>> str(x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character u'\xb2' in position 0: ordinal not in range(128)
>>> '%s' % x
u'\xb2'
```

Reply:
Ah okay. I'll `%s` them and any `str()` in the future. Ty for the quick review.

Reply:
Yeah the unicode stuff is a pain.  I've just run into this sort of stuff enough times to have learnt my lesson.  But it is always good to assume the user will put unicode where you least expect it.  but as shown above you can usually test things quite easily.  The main thing is to be aware of the issue.  It all comes in time.

I'm working on getting some time to push through a load of the overdue PRs.  Also you should sign up to https://blog.digitalocean.com/hacktoberfest-2017/ if you want a free T-shirt etc

Reply:
>@lasers please be very careful changing things just because you can.

Because I like things to be consistent and I try to optimize things too. I recently omitted two `str()` (delta) in favor of doing it once in `post_config_hook`. Hopefully this helps a picobit.

>I'm working on getting some time to push through a load of the overdue PRs.

No rush. Nice of us to start addressing old issues. Let me know when and I'll get on IRC so we can discuss/address/review things... and start merging things out of the way. The numbers are going to drop down rapidly.